release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+  }
+
+  private void categorizeReleaseNoteText(ReleaseNoteText rnText) {
+    checkReleaseNoteCategory(rnText);
+    Arrays.stream(Category.values())
+        .forEach(
+            category -> {
+              if (rnText.category.equals(category)) {
+                List<ReleaseNoteText> releaseNoteTexts =
+                    categoryMap.computeIfAbsent(category, k -> new ArrayList<>());
+                if (!isContainedInSameAsItems(rnText)) releaseNoteTexts.add(rnText);
+                categoryMap.put(category, releaseNoteTexts);
+              }
+            });
+  }
+
+  void assortSameAsItems() {
+    for (Entry<String, List<ReleaseNoteText>> entry : sameAsItems.entrySet()) {
+      String topicPrNumber = entry.getKey();
+      List<ReleaseNoteText> releaseNoteTextsInSameAs = entry.getValue();
+
+      Arrays.stream(Category.values())
+          .forEach(
+              category -> {
+                List<ReleaseNoteText> releaseNoteTextsInACategory = categoryMap.get(category);
+                if (releaseNoteTextsInACategory != null) {
+                  releaseNoteTextsInACategory.forEach(
+                      rnInfo -> {
+                        if (rnInfo.prNumbers.get(0).equals(topicPrNumber)) {
+                          releaseNoteTextsInSameAs.forEach(from -> merge(from, rnInfo));
+                        }
+                      });
+                }
+              });
+    }
+  }
+
+  private void merge(ReleaseNoteText from, ReleaseNoteText to) {
+    if (from.text != null && !from.text.isEmpty()) {
+      to.text = to.text + " " + from.text;
+    }
+    if (DEBUG != null) System.err.printf("merged RN text: %s%n", to.text);
+    to.prNumbers.addAll(from.prNumbers);
+  }
+
+  void outputReleaseNote() {
+    StringBuilder builder = new StringBuilder();
+    builder.append("## Summary\n\n");
+
+    Arrays.stream(Category.values())
+        .forEach(
+            category -> {
+              List<ReleaseNoteText> releaseNotes = categoryMap.get(category);
+              if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                builder.append(String.format("## %s%n", category.getDisplayName()));
+                builder.append(getFormattedReleaseNotes(releaseNotes)).append("\n");
+              }
+            });
+    System.out.println(builder);
+  }
+
+  private String getFormattedReleaseNotes(List<ReleaseNoteText> releaseNotes) {
+    StringBuilder builder = new StringBuilder();
+    releaseNotes.forEach(
+        rnText -> {
+          builder.append(String.format("- %s (", rnText.text));
+          rnText.prNumbers.forEach(prNum -> builder.append(String.format("#%s ", prNum)));
+          builder.deleteCharAt(builder.length() - 1); // delete the last space character
+          builder.append(")\n");
+        });
+    return builder.toString();
+  }
+
+  private void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+    if (rnText.category == null) {
+      rnText.category = Category.MISCELLANEOUS;
+    }
+  }
```

Comment:
IMO, since it does more than just checking the category, the name should be revised.

Reply:
Ah, yes. I'll revise the method name. Thank you.

Reply:
Done. 06385a5